ompi/mca/topo/base/topo_base_lazy_init.c

```diff
@@ -39,19 +41,25 @@ int mca_topo_base_lazy_init(void)
     int err;
 
     if (!mca_base_framework_is_open (&ompi_topo_base_framework)) {
-        /**
-         * Register and open all available components, giving them a chance to access the MCA parameters.
-         */
+        OPAL_THREAD_LOCK(&ompi_mpi_topo_bootstrap_mutex);
```

Comment:
I would rather have this safety moved down into the framework itself, as I don't think that any lazily initialized framework support, or need, to allow threaded initialization.

Reply:
Can you expand on this a bit? The io framework lazy init does something similar to what's done here.

Reply:
As you mentionned we have several frameworks that need to be initialized lazily potentially in a multi-threaded way. I would like to have a consistent way to handle their lazy initialization, to avoid any discrepancies between them.

Reply:
I opened https://github.com/open-mpi/ompi/issues/9307 for discussing the lazy framework initialization. I can work on that once we have consensus. We can remove this specific commit from the PR, so that the rest needn't be held up.

Reply:
Thanks @nysal . Will remove the commit.